### PR TITLE
Notification rule revert

### PR DESCRIPTION
**Pull request by jozefbakus**
Implements #583 - Revert notification rule

### CODE REVIEW

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/NotificationRuleHistoryController.scala [hunk 1]
```diff
@@ -32,6 +32,11 @@ class NotificationRuleHistoryController @Inject()(notificationRuleHistoryService
     notificationRuleHistoryService.getHistoryForNotificationRule(notificationRuleId).toJava.toCompletableFuture
   }
 
+  @GetMapping(path = Array("/historyNotificationRule"))
+  def getHistoryNotificationRule(@RequestParam notificationRuleHistoryId: Long): CompletableFuture[NotificationRule] = {
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/NotificationRuleHistoryController.scala]: I'd rename this endpoint to `/notificationRuleFromHistory` to stay consistent

File: ui/src/app/components/admin/notification-rules/notification-rule-history/notification-rule-comparison/notification-rule-comparison.component.html [hunk 2]
```diff
@@ -22,6 +22,13 @@
 <div class="content-area" *ngIf="!loading">
   <div class="clr-row" *ngIf="isLoadedSuccessfully()">
     <div class="clr-col oneSideOfComparison" >
+      <div class="btn-group">
+        <button type="button" class="btn btn-sm btn-secondary"
+                [routerLink]="[absoluteRoutes.REVERT_NOTIFICATION_RULE, leftHistory.history.id]">
+          <clr-icon shape="redo"></clr-icon>
+          Revert
```
Comment (kevinwallimann) [inline ui/src/app/components/admin/notification-rules/notification-rule-history/notification-rule-comparison/notification-rule-comparison.component.html]: Maybe "Revert to this version"?

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/NotificationRuleHistoryController.scala (hunk 1 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/NotificationRuleHistoryController.scala]: Fixed

File: ui/src/app/components/admin/notification-rules/notification-rule-history/notification-rule-comparison/notification-rule-comparison.component.html (hunk 2 above)
Comment (jozefbakus) [reply ui/src/app/components/admin/notification-rules/notification-rule-history/notification-rule-comparison/notification-rule-comparison.component.html]: Fixed

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**